pkg/statistics/handle/globalstats/global_stats_internal_test.go

```diff
@@ -351,6 +351,7 @@ func testIssues24349(testKit *testkit.TestKit) {
 	testKit.MustExec("create table t (a int, b int) partition by hash(a) partitions 3")
 	testKit.MustExec("insert into t values (0, 3), (0, 3), (0, 3), (0, 2), (1, 1), (1, 2), (1, 2), (1, 2), (1, 3), (1, 4), (2, 1), (2, 1)")
 	testKit.MustExec("analyze table t with 1 topn, 3 buckets")
```

Comment:
Should we add some new test about the pruning logic in smaller table ? And check the topn number which has been pruned ?

Reply:
Added a testcase to TestPruneTopN function in statistics_test.go